pkg/scheduler/actions/shuffle/shuffle.go

```diff
@@ -0,0 +1,73 @@
+/*
+ Copyright 2022 The Volcano Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package shuffle
+
+import (
+	"k8s.io/klog"
+	"volcano.sh/volcano/pkg/scheduler/api"
+	"volcano.sh/volcano/pkg/scheduler/framework"
+)
+
+const Strategies = "strategies"
+
+// Action defines the action
+type Action struct{}
+
+// New returns the action instance
+func New() *Action {
+	return &Action{}
+}
+
+// Name returns the action name
+func (alloc *Action) Name() string {
+	return "shuffle"
+}
+
+// Initialize inits the action
+func (alloc *Action) Initialize() {}
+
+// Execute select evictees according given strategies and evict them.
+func (alloc *Action) Execute(ssn *framework.Session) {
+	klog.V(3).Infof("Enter Shuffle ...\n")
+	defer klog.V(3).Infof("Leaving Shuffle ...\n")
+
+	// select pods that may be evicted
+	tasks := make([]*api.TaskInfo, 0)
+	for _, jobInfo := range ssn.Jobs {
+		for _, taskInfo := range jobInfo.Tasks {
+			if taskInfo.Status == api.Running {
+				tasks = append(tasks, taskInfo)
+			}
+		}
+	}
+	for _, task := range tasks {
+		klog.V(4).Infof("Running tasks %s: [ns: %s, job: %s]\n", task.Name, task.Namespace, task.Job)
+	}
+
+	// Evict target workloads
+	victims := ssn.Victims(tasks)
+	for victim, _ := range victims {
+		klog.V(4).Infof("Victim %s: [ns: %s, job: %s]\n", victim.Name, victim.Namespace, victim.Job)
+		if err := ssn.Evict(victim, "shuffle"); err != nil {
```

Comment:
let's have a const value for `"shuffle"`.

Reply:
I'm OK about that.

Reply:
resolved